atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/compact/BackgroundCompactor.java

```diff
@@ -49,40 +50,57 @@
     private final TransactionManager transactionManager;
     private final KeyValueService keyValueService;
     private final LockService lockService;
-    private final Supplier<CompactorConfig> config;
+    private final Supplier<Boolean> inSafeHours;
 
     private Thread daemon;
 
-    public static void createAndRun(TransactionManager transactionManager,
+    public static Optional<BackgroundCompactor> createAndRun(TransactionManager transactionManager,
             KeyValueService keyValueService,
             LockService lockService,
-            Supplier<CompactorConfig> compactorConfigSupplier) {
+            Supplier<Boolean> inSafeHours) {
+        if (!keyValueService.shouldManuallyCompact()) {
+            return Optional.empty();
+        }
```

Comment:
To be honest for this kind of thing I usually prefer a null object pattern (basically you always create-and-run a background compactor, just that if your KVS shouldn't manually compact then the compactor does nothing) as it's easier to manage.

I realise it's quite a significant break from the existing API, though. Not actionable, to be clear.

Reply:
Ack and agreed, but will probably defer this refactor, especially as I don't think it would break anyone if we switched it around later.